tests/profile/test_jsonschema_profiler.py

```diff
@@ -817,6 +1093,71 @@ def test_has_profile_create_expectations_from_complex_schema(
             },
             "meta": {},
         },
+        {
+            "expectation_type": "expect_column_values_to_not_be_null",
+            "kwargs": {"column": "country-name"},
+            "meta": {},
+        },
+    ]
+    context = empty_data_context
+    context.save_expectation_suite(obs)
+
+
+def test_null_fields_schema(empty_data_context, null_fields_schema):
+    profiler = JsonSchemaProfiler()
+    obs = profiler.profile(null_fields_schema, "null_fields")
+    assert isinstance(obs, ExpectationSuite)
+    assert obs.expectation_suite_name == "null_fields"
+    assert [e.to_json_dict() for e in obs.expectations] == [
+        {
+            "meta": {},
+            "expectation_type": "expect_column_to_exist",
+            "kwargs": {"column": "null"},
+        },
+        {
+            "expectation_type": "expect_column_values_to_be_null",
+            "kwargs": {"column": "null"},
+            "meta": {},
+        },
+        {
+            "meta": {},
+            "expectation_type": "expect_column_to_exist",
+            "kwargs": {"column": "string-or-null"},
+        },
+        {
+            "meta": {},
+            "expectation_type": "expect_column_values_to_be_in_type_list",
+            "kwargs": {
+                "column": "string-or-null",
+                "type_list": list(ProfilerTypeMapping.STRING_TYPE_NAMES),
+            },
+        },
```

Comment:
I might be misunderstanding the jsonschema spec here:
```
"string-or-null": {"type": ["string", "null"]},
"int-or-null": {"type": ["integer", "null"]},
"number-or-null": {"type": ["number", "null"]},
```
but I would expect an `expect_column_values_to_not_be_null` for all of the `*-or-null` columns. Or am I reading the jsonschema incorrectly?

Reply:
Ok, after considerable discussion my comments above have been proven incorrect because GE ignores null values by default so no additional expectation is required to encode this assertion.